colossalai/shardformer/layer/attn.py

```diff
@@ -471,19 +472,50 @@ def get_double_ring_groups(sp_group, inner_ring_size=None):
         inner_ring_group = None
         inter_ring_group = None
 
-        # Create inner ring groups
-        for i in range(inner_ring_size):
-            ranks = list(range(i * inner_ring_size, (i + 1) * inner_ring_size))
-            group = dist.new_group(ranks)
-            if sp_rank in ranks:
-                inner_ring_group = group
+        world_size = dist.get_world_size()
+        rank = dist.get_rank()
+
+        num_ring_size = world_size // num_rings
+
+        if tp_size > 1:
+            # Create inner ring groups
+            ranks = []
+            for i in range(num_rings):
+                start = i * num_ring_size
+                end = (i + 1) * num_ring_size
+                for idx in range(start, end):
+                    inner_rank = []
+                    for k in range(inner_ring_size):
+                        current_num = idx + k * tp_size
+                        if current_num >= end:
+                            break
+                        inner_rank.append(current_num)
```

Comment:
Personally I feel it's more concise to use list comprehension here

Reply:
Okay

Reply:
fixed